File: php/resistor-color-duo/ResistorColorDuo.php
<?php

const BANDS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];

class ResistorColorDuo {
    public function getColorsValue(array $colors): int {
        return (int) implode('', array_map(fn($c) => array_search(strtolower($c), BANDS), array_slice($colors, 0, 2)));
    }
}

File: fsharp/resistor-color-duo/ResistorColorDuo.fs
module ResistorColorDuo
let colors = ["black"; "brown"; "red"; "orange"; "yellow"; "green"; "blue"; "violet"; "grey"; "white"]

let getIndex (target: string) = colors |> List.findIndex(fun colo -> colo = target)
let value (colors: List<string>) = (colors[0] |> getIndex) * 10 + (colors[1] |> getIndex)
